### DISCUSSION TITLE
UnknownHostException: kafka-zookeeper-client: Name or service not known

### DISCUSSION THREAD

**Discussion by mausch** [strimzi/strimzi-kafka-operator#9339] [Q&A] [answered]
Hi, I installed https://github.com/strimzi/strimzi-kafka-operator/blob/0.38.0/examples/kafka/kafka-persistent-single.yaml 
Zookeeper runs just fine.
Broker crashloops with "UnknownHostException: kafka-zookeeper-client: Name or service not known"
The broker config (displayed when the pod starts up) has `zookeeper.connect=kafka-zookeeper-client:2181`. However the zookeeper pod's name is `kafka-zookeeper-0` . Any ideas?

**Comment by ppatierno** [comment]
The broker configuration is right, because brokers are using a Kubernetes `Service` named like `<kafka_cluster_name>-zookeeper-client` to connect to ZooKeeper nodes as clients. I would check if you have any DNS issue in your environment because the ZK client is not able to resolve the host.

**Comment by mausch** [comment]
Thanks! Right, yes, just noticed that. Service `kafka-zookeeper-client` is bound fine to pod `kafka-zookeeper-0`. I can resolve it just fine from other pods too... 🤷‍♂️

**Reply by scholzj** [reply, marked as answer]
I guess you would need to look into your DNS and Kubernetes cluster. They are responsible for DNS resolution. I have no idea what environment do you use, but this can be also sometimes faulty juts on one worker node and when you start a pod on another node it works fine. You can also try to delete the pod and have it recreated in case it was just some temporary glitch.

**Reply by mausch** [reply]
Fixed, it was a faulty security group in one nodegroup which prevented talking to  CoreDNS pods in a different nodegroup.
